From: Danielle Ratson <danieller@mellanox.com>
Date: Thu, 17 Oct 2019 09:55:16 +0300
Subject: selftests: mlxsw: Add Spectrum-2 mirror-to-gretap target scale test
Patch-mainline: v5.5-rc1
Git-commit: cb7d2c719c288dee9ef0e97c66f404600795c7f9
References: bsc#1176774

Like in Spectrum, use the number of analyzers taken from the devlink
command.

Signed-off-by: Danielle Ratson <danieller@mellanox.com>
Acked-by: Jiri Pirko <jiri@mellanox.com>
Signed-off-by: Ido Schimmel <idosch@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 tools/testing/selftests/drivers/net/mlxsw/spectrum-2/mirror_gre_scale.sh |   16 ++++++++++
 1 file changed, 16 insertions(+)
 create mode 100644 tools/testing/selftests/drivers/net/mlxsw/spectrum-2/mirror_gre_scale.sh

--- /dev/null
+++ b/tools/testing/selftests/drivers/net/mlxsw/spectrum-2/mirror_gre_scale.sh
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: GPL-2.0
+source ../mirror_gre_scale.sh
+
+mirror_gre_get_target()
+{
+	local should_fail=$1; shift
+	local target
+
+	target=$(devlink_resource_size_get span_agents)
+
+	if ((! should_fail)); then
+		echo $target
+	else
+		echo $((target + 1))
+	fi
+}
